tests/components/sensor/test_pilight.py

```diff
@@ -0,0 +1,120 @@
+"""The tests for the Pilight sensor platform."""
+import logging
+
+from homeassistant.bootstrap import setup_component
+import homeassistant.components.sensor as sensor
+from homeassistant.components import pilight
+
+from tests.common import get_test_home_assistant, assert_setup_component
+
+HASS = None
+
+
+def fire_pilight_message(protocol, data):
+    """Fire the fake pilight message."""
+    message = {pilight.ATTR_PROTOCOL: protocol}
+    message.update(data)
+    HASS.bus.fire(pilight.EVENT, message)
+
+
+def setUpModule():   # pylint: disable=invalid-name
+    """Initialize a Home Assistant server."""
+    global HASS
+
+    HASS = get_test_home_assistant()
+    HASS.config.components = ['pilight']
+
+
+def tearDownModule():   # pylint: disable=invalid-name
+    """Stop the Home Assistant server."""
+    HASS.stop()
+
+
+def test_sensor_value_from_code():
+    """Test the setting of value via pilight."""
+    with assert_setup_component(1):
```

Comment:
The first parameter to `assert_setup_component` is a `len` of the returned platforms or the length of the component's config dictionary. So in the case of a component you can adjust the 1 to suit the amount of keys in the dictionary.

Another way to test further is to check the returned config

```python
with assert_component_setup(xx) as config:
  setup_component()
  assert isinstance(config, dict)
```

Unfortunately the number is not optional at the moment, but you should be able to get around it.